test/platform/test-device-manager.ts

```diff
@@ -0,0 +1,125 @@
+import test from 'ava';
+import {
+  DeviceManagerStub,
+  ERROR_UNEXPECTED_COMMAND_REQUEST,
+  ERROR_PENDING_REQUEST_MISMATCH,
+  UdpResponseData,
+} from '../../src';
+import {createUdpDeviceCommand} from '../example/fixtures';
+
+const EXECUTE_REQUEST_ID = 'execute-request-id';
+const DEVICE_ID = 'test-device-id';
+const COMMAND_REQUEST = createUdpDeviceCommand(
+  Buffer.from('test-execute-buffer'),
+  EXECUTE_REQUEST_ID,
+  DEVICE_ID,
+  12345
+);
+
+/**
+ * Returns a simple execute request used for testing
+ * @param command A command to set in the sample request
+ */
+function createExecuteRequest(command: string) {
+  return {
+    requestId: EXECUTE_REQUEST_ID,
+    inputs: [
+      {
+        intent: smarthome.Intents.EXECUTE,
+        payload: {
+          commands: [
+            {
+              execution: [
+                {
+                  command,
+                  params: {},
+                },
+              ],
+              devices: [
+                {
+                  id: DEVICE_ID,
+                },
+              ],
+            },
+          ],
+          structureData: {},
+        },
+      },
+    ],
+  };
+}
+
+/**
+ * Tests that `markPending()` matches two identical requests
+ */
+test('device-manager-expected-mark-pending', async t => {
+  const deviceManager = new DeviceManagerStub();
+  const executeReqeust = createExecuteRequest('action.devices.commands.OnOff');
```

Comment:
`executeRequest`

Reply:
Don't we need parameters along with a command?

Reply:
Since this test is just differentiating any two non-identical execute requests, it's not really required here, but I added it for completion